src/Microsoft.DotNet.ProjectJsonMigration/Rules/AddDefaultsToProjectRule.cs

```diff
@@ -0,0 +1,123 @@
+            Dictionary<string, ProjectItemGroupElement> defaultsItemGroups)
+        {
+            foreach (var itemInfo in defaults.Items)
+            {
+                ProjectItemGroupElement itemGroup;
+                var parentCondition = itemInfo.ParentCondition ?? string.Empty;
+
+                if (!defaultsItemGroups.TryGetValue(parentCondition, out itemGroup))
+                {
+                    itemGroup = project.AddItemGroup();
+                    itemGroup.Label = c_DefaultsProjectElementContainerLabel;
+                    itemGroup.Condition = parentCondition;
+
+                    defaultsItemGroups[parentCondition] = itemGroup;
+                }
+
+                var item = itemGroup.AddItem(itemInfo.ItemType, itemInfo.Include);
+                item.Exclude = itemInfo.Exclude;
+                item.Remove = itemInfo.Remove;
+                item.Condition = itemInfo.Condition;
+            }
+        }
+
+        private static void AddDefaultPropertiesToProject(
+            SerializableMigrationDefaultsInfo defaults, 
+            ProjectRootElement project, 
+            Dictionary<string, ProjectPropertyGroupElement> defaultsPropertyGroups)
+        {
+            foreach (var propertyInfo in defaults.Properties)
+            {
+                ProjectPropertyGroupElement propertyGroup;
+                var parentCondition = propertyInfo.ParentCondition ?? string.Empty;
+
+                if (!defaultsPropertyGroups.TryGetValue(parentCondition, out propertyGroup))
+                {
+                    propertyGroup = project.AddPropertyGroup();
+                    propertyGroup.Label = c_DefaultsProjectElementContainerLabel;
+                    propertyGroup.Condition = parentCondition;
+
+                    defaultsPropertyGroups[parentCondition] = propertyGroup;
+                }
+
+                var property = propertyGroup.AddProperty(propertyInfo.Name, propertyInfo.Value);
+                property.Condition = propertyInfo.Condition;
+            }
+        }
+
+        private SerializableMigrationDefaultsInfo ResolveDefaults()
+        {
+            var sdkDefaultFile = TryResolveSdkDefaultsFileFromAppBase();
+            if (sdkDefaultFile != null)
+            {
+                return DeserializeDefaults(File.ReadAllText(sdkDefaultFile));
+            }
+
+            var thisAssembly = typeof(AddDefaultsToProjectRule).GetTypeInfo().Assembly;
+            using (var resource = thisAssembly.GetManifestResourceStream("Microsoft.DotNet.ProjectJsonMigration." + c_SdkDefaultsJsonFileName))
+            {
+                using (StreamReader reader = new StreamReader(resource))
+                {
+                    return DeserializeDefaults(reader.ReadToEnd());
+                }
+            }
+        }
+
+        private string TryResolveSdkDefaultsFileFromAppBase()
+        {
+            var candidate = Path.Combine(AppContext.BaseDirectory, c_SdkDefaultsJsonFileName);
```

Comment:
Is it weird to put "configuration" type things next to the CLI?  The CLI could be installed in a location that isn't writeable by the user, right?  What if we added this as an option to the `dotnet migrate` command?